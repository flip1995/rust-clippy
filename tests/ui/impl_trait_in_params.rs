#![allow(unused)]
#![warn(clippy::impl_trait_in_params)]

pub trait Trait {}

// Should warn
pub fn a(_: impl Trait) {}
pub fn c<C: Trait>(_: C, _: impl Trait) {}

// Shouldn't warn

pub fn b<B: Trait>(_: B) {}
fn d<D: Trait>(_: D, _: impl Trait) {}

fn main() {}
