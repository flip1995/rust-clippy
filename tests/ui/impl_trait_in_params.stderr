error: 'impl Trait' in the function's parameters
  --> $DIR/impl_trait_param.rs:7:13
   |
LL | pub fn a(_: impl Trait) {}
   |             ^^^^^^^^^^
   |
   = note: `-D clippy::impl-trait-param` implied by `-D warnings`
help: create a generic type here and replace that 'impl Trait' with `T`
   |
LL | pub fn a<T: Trait>(_: impl Trait) {}
   |         ++++++++++

error: 'impl Trait' in the function's parameters
  --> $DIR/impl_trait_param.rs:8:29
   |
LL | pub fn c<C: Trait>(_: C, _: impl Trait) {}
   |                             ^^^^^^^^^^
   |
help: create a generic type here and replace that `impl Trait` with `T`
   |
LL | pub fn c<C: Trait, T: Trait>(_: C, _: impl Trait) {}
   |                  ++++++++++

error: aborting due to 2 previous errors

